apps/web/src/components/invoice-generator/index.tsx

```diff
@@ -0,0 +1,478 @@
+        <span>
+          <InputRow
+            label={t("Unit price")}
+            type="number"
+            name="rows.unit_price"
+            id={`rows[${index.toString()}].unit_price`}
+            required
+            step={0.01}
+            min={0}
+            unit="€"
+          />
+        </span>
+      </ErrorMessageBlock>
+    </fieldset>
+  );
+}
+
+function AttachmentRow({
+  state,
+  index,
+}: {
+  state: InvoiceGeneratorFormState | null;
+  index: number;
+}) {
+  const t = useScopedI18n("invoicegenerator");
+
+  return (
+    <fieldset>
+      <ErrorMessageBlock
+        elementName={`attachment_descriptions[${index.toString()}]`}
+        formState={state}
+      >
+        <TextAreaInputRow
+          label={t("Description")}
+          id={`attachment_descriptions[${index.toString()}]`}
+          name="attachment_descriptions"
+          maxLength={512}
+          required
+        />
+      </ErrorMessageBlock>
+      <ErrorMessageBlock
+        elementName={`attachments[${index.toString()}]`}
+        formState={state}
+      >
+        <InputRow
+          label={t("Attachment")}
+          id={`attachments[${index.toString()}]`}
+          name="attachments"
+          type="file"
+          required
+        />
+      </ErrorMessageBlock>
+    </fieldset>
+  );
+}
+
+function InvoiceGeneratorForm() {
+  const [state, formAction] = useFormState(SaveAction, null);
+  const t = useScopedI18n("invoicegenerator");
+
+  useEffect(() => {
+    const errorFields = state?.errors ? Object.keys(state.errors) : [];
+    if (errorFields.length !== 0) {
+      const lastErrorField = errorFields[errorFields.length - 1];
+      const inputWithError =
+        (document.getElementsByName(lastErrorField)[0] as HTMLElement | null) ??
+        document.getElementById(lastErrorField);
+      inputWithError?.focus();
+    } else {
+      document.querySelector("[data-form-status]")?.scrollIntoView();
+    }
+  }, [state]);
+
+  return (
+    <form
+      className="shadow-solid w-full max-w-prose space-y-4 overflow-x-clip rounded-md border-2 border-gray-900 p-4 py-6 md:px-6 md:py-8"
+      action={formAction}
+    >
+      <label htmlFor="recipient-info">
+        <h2>{t("Invoicer information")}</h2>
+      </label>
+
+      <fieldset id="recipient-info">
+        <ErrorMessageBlock elementName="recipient_name" formState={state}>
+          <InputRow
+            label={t("Invoicer name")}
+            name="recipient_name"
+            autoComplete="name"
+            maxLength={128}
+            required
+          />
+        </ErrorMessageBlock>
+        <ErrorMessageBlock elementName="recipient_email" formState={state}>
+          <InputRow
+            label={t("Invoicer email")}
+            maxLength={128}
+            name="recipient_email"
+            type="email"
+            required
+          />
+        </ErrorMessageBlock>
+        <ErrorMessageBlock elementName="phone_number" formState={state}>
+          <InputRow
```

Comment:
I think we should add some sort of hint about the format for better UX, first step could be valid number as placeholder, next step dropdown with country codes for example :thinking: